server/authentication/authentication.js

```diff
@@ -0,0 +1,42 @@
+const { sign, verify } = require('jsonwebtoken');
+require('env2')('config.env');
+
+const createCookie = (user, cb) => {
+  sign(user, process.env.SECRET, (signerror, token) => {
+    if (signerror) { cb(signerror); } else {
+      cb(null, token);
+    }
+  });
+};
+
+const getTokenData = (data, cb) => {
+  verify(data, process.env.SECRET, (err, decoded) => {
+    if (err) {
+      return cb(new TypeError());
+    }
+    cb(null, decoded);
+    return true;
+  });
+};
+
+const authCheck = (request, cb) => {
+  if (!request.cookies) {
+    return cb(new TypeError());
```

Comment:
Specify the error, here you will just catch the  error with no  specific message and that  isn't good for debugging